shared/src/business/entities/trialSessions/TrialSession.ts

```diff
@@ -573,4 +573,5 @@ export const TrialSession: typeof TrialSessionClass =
   validEntityDecorator(TrialSessionClass);
 
 // eslint-disable-next-line no-redeclare
-export interface TrialSessionClass extends TValidationEntity {}
+export interface TrialSessionClass
```

Comment:
I like the idea that our entities all implement some validationEntity so we know the common validation functions like .validate().

This interface is a little confusing to me though as we are exporting an interface called "TrialSessionClass" and it is not a class. Also higher in this file we are exporting a class called "TrialSessionClass". 

I think what you may be looking for is a single class that implements TValidationEntity. So something like:

export class TrialSessionClass implements TValidationEntity<whateverType> {
constructor()
validate()
blah()
}

Reply:
I did originally try this approach, but then our entities light up red unless we implement stub methods for all those required interface's method which we end up overwriting later on which I felt was more dirty than my current approach.  The reason for the class TrialSessionClass and the interface TrialSessionClass is that typescript actually merges both of those interfaces together which allows consumers of this file to know that a TrialSessionClass is actually a combination of the class and the joi validation methods together.